File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/Effects.vb
Imports DevExpress.Xpf.Charts
Imports System.ComponentModel.DataAnnotations

Namespace PropertyGridDemo

    <MetadataType(GetType(Metadata.MirrorOptionsMetadata))>
    Public Class MirrorOptions

        Public Sub New()
            MirrorLength = 50
        End Sub

        Public Overridable Property MirrorLength As Double

        Public Overridable Property Show As Boolean

        Public Overridable Property ActualLength As Double

        Protected Sub OnMirrorLengthChanged()
            UpdateActualLength()
        End Sub

        Protected Sub OnShowChanged()
            UpdateActualLength()
        End Sub

        Private Sub UpdateActualLength()
            If Show Then
                ActualLength = MirrorLength
            Else
                ActualLength = 0R
            End If
        End Sub
    End Class

    <MetadataType(GetType(Metadata.LabelOptionsMetadata))>
    Public Class LabelOptions

        Public Overridable Property ShowLabel As Boolean

        Public Overridable Property Position As Bar2DLabelPosition
    End Class

    <MetadataType(GetType(Metadata.VisibleLabelOptionsMetadata))>
    Public Class VisibleLabelOptions
        Inherits LabelOptions

    End Class
End Namespace

File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/Effects.cs
using DevExpress.Xpf.Charts;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PropertyGridDemo {    
    [MetadataType(typeof(Metadata.MirrorOptionsMetadata))]
    public class MirrorOptions {
        public MirrorOptions() {
            MirrorLength = 50;
        }
        public virtual double MirrorLength { get; set; }
        public virtual bool Show { get; set; }
        public virtual double ActualLength { get; set; }

        protected void OnMirrorLengthChanged() {
            UpdateActualLength();
        }        
        protected void OnShowChanged() {
            UpdateActualLength();
        }
        void UpdateActualLength() {
            if (Show)
                ActualLength = MirrorLength;
            else
                ActualLength = 0d;            
        }
    }
    [MetadataType(typeof(Metadata.LabelOptionsMetadata))]
    public class LabelOptions {
        public virtual bool ShowLabel { get; set; }
        public virtual Bar2DLabelPosition Position { get; set; }
    }
    [MetadataType(typeof(Metadata.VisibleLabelOptionsMetadata))]
    public class VisibleLabelOptions : LabelOptions {        
    }
}
